pkg/entities/analysis/analysis.go

```diff
@@ -145,3 +151,7 @@ func (a *Analysis) GetDataWithoutVulnerabilities() *Analysis {
 		FinishedAt:     a.FinishedAt,
 	}
 }
+
+func (a *Analysis) AddWarning(warning string) {
+	a.Warnings = append(a.Warnings, warning)
+}
```

Comment:
Since the Analysis object is shared between all formatters that are executed in parallel this code can result in data races.


By the way, I don't think this approach would be good. I don't like the idea of sharing the Analysis object between all formatters and adding one more field that makes this mandatory I don't think it would be good. Can we find a better approach to implement this Warnings?

Maybe change the formatter to return a list of warning's? Implement a custom logger that any call to logger.Warn is printed only after the analysis is finished? What do you guys think? @wiliansilvazup @iancardosozup

Reply:
My idea was not to share the object of analysis between the formatters, as we talked during a meeting, the formatters will only be responsible for parsing and returning the necessary data, warnings included. So this field for the formatters, would only be used after refactoring.

Reply:
If this field will only be used after refactor of formatters I think that should not be included in this pr.

Reply:
I think I was not very clear, correcting. In the case of formatters, this field would only be used for refactoring. But I already use this field in the cli to present the hashes warnings.

Reply:
I see that the idea of @matheusalcantarazup fits better in the solution to remove the devkit. In my opinion for the problem we have here this is an implementation.